pkg/controllers/member/serviceexport/controller.go

```diff
@@ -311,30 +320,60 @@ func (r *Reconciler) setAzureRelatedInformation(ctx context.Context, service *co
 		// to avoid sending Azure requests.
 		return nil
 	}
-	export.Spec.PublicIPResourceID = pip.ID
+	hubSvcExport.Spec.PublicIPResourceID = pip.ID
 
 	// Note the user can set the dns label via the Azure portal or Azure CLI without updating service.
 	// This information may be stale as we don't monitor the public IP address resource.
-	export.Spec.IsDNSLabelConfigured = pip.Properties != nil && pip.Properties.DNSSettings != nil && pip.Properties.DNSSettings.DomainNameLabel != nil
+	hubSvcExport.Spec.IsDNSLabelConfigured = pip.Properties != nil && pip.Properties.DNSSettings != nil && pip.Properties.DNSSettings.DomainNameLabel != nil
 
 	// No matter if the customer bring your own IP or not, the cloud provider will reconcile the DNS label based on the
 	// DNS annotation.
 	dnsName, found := service.Annotations[objectmeta.ServiceAnnotationAzureDNSLabelName]
-	klog.V(2).InfoS("Finding whether the DNS is assigned", "service", serviceKObj, "dnsName", dnsName, "isSetOnService", found, "isConfiguredOnPIP", export.Spec.IsDNSLabelConfigured)
+	klog.V(2).InfoS("Finding whether the DNS is assigned", "service", serviceKObj, "dnsName", dnsName, "isSetOnService", found, "isConfiguredOnPIP", hubSvcExport.Spec.IsDNSLabelConfigured)
 	// If the annotation is not set, the cloud provider won't reconcile the DNS label and return the current status.
 	if !found {
 		// cloud provider won't delete DNS label on pip if the annotation is not set.
 		return nil
 	}
 	if len(dnsName) == 0 {
-		export.Spec.IsDNSLabelConfigured = false // cloud provider will delete the DNS label on the pip.
+		hubSvcExport.Spec.IsDNSLabelConfigured = false // cloud provider will delete the DNS label on the pip.
 		return nil
 	}
-	if !export.Spec.IsDNSLabelConfigured {
+	if !hubSvcExport.Spec.IsDNSLabelConfigured {
 		err = fmt.Errorf("in the process of adding DNS to the public ip address %s", *pip.ID)
 		klog.ErrorS(err, "Requeue the request to see if the DNS is ready or not", "service", serviceKObj)
 		return err
 	}
+
+	return nil
+}
+
+// setWeightOnHubServiceExport sets the weight on the hub cluster's InternalServiceExport based on the weight annotation
+// set on the member cluster's ServiceExport.
+func setWeightOnHubServiceExport(svcExport *fleetnetv1alpha1.ServiceExport, hubSvcExport *fleetnetv1alpha1.InternalServiceExport) error {
```

Comment:
for the invalid weight, i'm thinking of defaulting to 1, instead of returning the error. Here the controller is keeping retry and the customer won't know what happened only after checking the logs.

Another option is to update the export condition as invalid to surface the cx error.